db/post_migrate/20220701195731_convert_chatable_topics_to_categories.rb

```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+class ConvertChatableTopicsToCategories < ActiveRecord::Migration[7.0]
+  def up
+    # convert chatable topics to categories using topic's category_id or default category
+    DB.exec(<<~SQL, uncategorized_category_id: SiteSetting.uncategorized_category_id)
+      UPDATE chat_channels cc
+      SET (chatable_type, chatable_id, name) = (
+        SELECT 'Category', coalesce(t.category_id, :uncategorized_category_id), coalesce(cc.name, t.title)
+        FROM topics t
+        WHERE cc.chatable_id = t.id
+      )
+      WHERE cc.chatable_type = 'Topic'
+      SQL
+
+    # soft delete all posts small actions
+    DB.exec(
+      "UPDATE posts SET deleted_at = :deleted_at, deleted_by_id = :deleted_by WHERE action_code IN (:action_codes)",
+      action_codes: ['chat.enabled', 'chat.disabled'],
+      deleted_at: Time.zone.now,
+      deleted_by: Discourse.system_user
+    )
```

Comment:
I'd rather use an explicit ID instead of relying on Rails magic, but the current SQL works, so it's up to you. :man_shrugging: 

```suggestion
    DB.exec(
      "UPDATE posts SET deleted_at = :deleted_at, deleted_by_id = :deleted_by_id WHERE action_code IN (:action_codes)",
      action_codes: ['chat.enabled', 'chat.disabled'],
      deleted_at: Time.zone.now,
      deleted_by_id: Discourse::SYSTEM_USER_ID
    )
```